messaging/client.go

```diff
@@ -102,49 +102,40 @@ func (c *Client) Open(path string, urls []*url.URL) error {
 		return ErrClientOpen
 	}
 
-	var seedUrls []*url.URL
-
-	// Determine which seed URLs to use.
-	if len(urls) > 0 {
-		seedUrls = urls
-	} else {
+	// Read URLs from file if no URLs are provided.
+	if len(urls) == 0 {
 		// Read URLs from config file. There is no guarantee
 		// that the Brokers URLs in the config file are still
 		// the Brokers, so we're going to double-check.
 		b, err := ioutil.ReadFile(path)
-		if err != nil {
-			return err
-		}
-		if err := json.Unmarshal(b, &c.config); err != nil {
+		if os.IsNotExist(err) {
+			// nop
+		} else if err != nil {
 			return err
+		} else {
+			if err := json.Unmarshal(b, &c.config); err != nil {
+				return err
+			}
+			urls = c.config.Brokers
 		}
-		seedUrls = c.config.Brokers
 	}
 
-	if len(seedUrls) < 1 {
+	if len(urls) < 1 {
 		return ErrBrokerURLRequired
 	}
 
 	// Now that we have the seed URLs, actually use these to
 	// get the actual Broker URLs. Do that here.
-	c.config.Brokers = seedUrls // Let's pretend they are the same
-
-	// Write the Broker URLs to disk.
-	b, err := json.Marshal(c.config.Brokers)
```

Comment:
Oh, I recall us discussing this -- we need to actually persist the URLs we get back from the cluster.